api/client/src/main/java/org/projectnessie/client/http/HttpClientBuilderImpl.java

```diff
@@ -279,7 +289,25 @@ public HttpClient build() {
       throw new IllegalArgumentException(
           "No HTTP client factory for name '" + clientName + "' found");
     }
-    return httpClientFactory.buildClient(config);
+    HttpClient client = httpClientFactory.buildClient(config);
+
+    if (authentication != null) {
+      try {
+        authentication.start();
+        if (cancellationCallbackConsumer != null) {
+          cancellationCallbackConsumer.accept(authentication::close);
+        }
+      } catch (RuntimeException e) {
+        try {
+          client.close();
+        } catch (RuntimeException e2) {
+          e.addSuppressed(e2);
+        }
+        throw e;
+      }
+    }
+
+    return client;
```

Comment:
Thinking a bit more about cancellation, I think you should close the `HttpClient` itself, not only the authentication:

```suggestion
    if (cancellationCallbackConsumer != null) {
      cancellationCallbackConsumer.accept(client::close);
    }
    return client;
```

Closing the HTTP client will close the authenticator.

Which actually begs the question: why not put the cancellation stuff at the `NessieApiV2` level? Closing the `NessieApiV2` instance will cascade-close everything underneath.